lib/rack/attack/store_proxy.rb

```diff
@@ -1,20 +1,26 @@
 # frozen_string_literal: true
 
+require 'delegate'
+
 module Rack
   class Attack
-    module StoreProxy
-      PROXIES = [
-        DalliProxy,
-        MemCacheStoreProxy,
-        RedisStoreProxy,
-        RedisProxy,
-        RedisCacheStoreProxy,
-        ActiveSupportRedisStoreProxy
-      ].freeze
+    class StoreProxy < SimpleDelegator
+      class << self
+        def proxies
+          @@proxies ||= []
```

Comment:
Would instance variable be enough here?

Reply:
No, should be class variable to work with chains of inheritance, like `class RedisProxy < StoreProxy`, `class RedisStoreProxy < RedisProxy`

Reply:
Gotcha.

Was thinking under the assumption all proxies inherited directly from `StoreProxy`, forgot that chained redis case.